.github/workflows/main.yml

```diff
@@ -0,0 +1,84 @@
+name: QA
+permissions:
+  contents: read
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-lock:
```

Comment:
fyi we've removed option to run only when tag is present on the PR:

```
 github.event_name == 'push' || (github.event.pull_request.draft == false || contains(github.event.pull_request.labels.*.name, 'Run QA checks'))
```

do we want to get rid of it?